gobblin-yarn/src/main/java/gobblin/yarn/GobblinWorkUnitRunner.java

```diff
@@ -174,6 +162,7 @@ public GobblinWorkUnitRunner(String applicationName, String helixInstanceName, C
       LOGGER.info("Adding YarnContainerSecurityManager since login is keytab based");
       services.add(new YarnContainerSecurityManager(config, fs, this.eventBus));
     }
+    services.add(new JMXReportingService());
```

Comment:
Why don't we use `ServiceBasedAppLauncher` for the services in the container as well?